### ISSUE TITLE
Show parsing & project symbol scanning progress in the status bar

### ISSUE THREAD

**Issue by hsandt** [slynch8/10x#2613]
I'm working with Unreal Engine 5 and it took quite a good amount of time (several minutes) to finish parsing all project symbols, enabling full Go To Symbol Definition navigation. Don't get me wrong, this is excellent as on my computer, I could barely make Visual Studio autocomplete work, while Rider is excellent at scanning everything but it's so zealous I would never end scanning and it would keep using my CPU.

10x Editor managed to do it in the end, but I wasn't able to tell that scanning was still in progress at first, so I thought it was a bug that Go To Symbol Definition failed to even find a method implementation from .h to .cpp. And when it finally worked, I couldn't say how much time it really required. Only by measuring the second time I launched it and spamming middle click on a method name could I confirm it was taking 15s (since editor launch) to parse my game classes (probably thx to the previous scan's cache), and I can't say what the initial parsing time was.

Showing a symbol scan / parsing progress bar in the status bar like Rider would help us know how much time was needed and from when we can start relying on symbol navigation.

**Comment by hsandt** [comment]
Interesting, looking at the introduction video: https://youtu.be/fH80sofUS7o?t=54 it seems that there is such a progress bar but in the top-right. I'll pay more attention to it next time I open a big project.

**Comment by slynch8** [comment]
Yes, there is a progress bar top right. You can see it by running the `ReparseAll` command. Unreal will take about 7 minutes to do a full reparse. After that it will just load the parser cache which should take about 10 seconds.

![image](https://github.com/slynch8/10x/assets/7141856/81fc23ff-1477-4b54-8d00-e712bcd229be)

I do want to make it more obvious that the parse is still in progress though, see https://github.com/slynch8/10x/issues/273
